BackendWebApp/src/main/java/uk/ac/ebi/biostd/webapp/application/security/service/GroupService.java

```diff
@@ -31,4 +34,15 @@ public UserGroup createGroup(UserGroup userGroup) {
     public List<UserGroup> getUsersGroups(long userId) {
         return new ArrayList<>(userRepository.getOne(userId).getGroups());
     }
+
+    public UserGroup getGroupFromUser(long userId, String groupName) {
+        return userGroupRepository.findByNameAndUsersContains(groupName, userRepository.getOne(userId))
+                .orElseThrow(() -> new ApiErrorException(String.format(
```

Comment:
I would suggest to create `EntityFoundException` at service layer, then we can mapped in controller advice, key point to do that is that api represent a rest layer exception and issue raised here is at service layer level.